include/gatekeeper_gk.h

```diff
@@ -122,6 +125,13 @@ struct gk_fib {
 			 */
 			struct ether_cache eth_cache;
 		} grantor;
+
+		/*
+		 * When the action is GK_FWD_BACK_NET or GK_FWD_FRONT_NET,
+		 * we need the next fib entry for the gateway of the
+		 * destination.
+		 */
+		struct gk_fib *next_fib;
```

Comment:
This comment refers to whole `struct gk_fib`. It is pointing to this line because I can only comment on what has changed.

Before adding BP and RT to GK, all neighbors and gateways were referring to neighbors and gateways in the back network. Now, the FIB needs to know on from which interface a neighbor or a gateway come, and the code must check that information where it makes sense, for example, if the GK block is acting as BP from back to front, the gateway returned by the LPM table must be from the front network.